pcdsdevices/mirror.py

```diff
@@ -0,0 +1,336 @@
+
+    def __init__(self, prefix, *, gantry_prefix=None, **kwargs):
+        self.gantry_prefix = gantry_prefix or 'GANTRY:' + prefix
+        self.follow_prefix = prefix + ':S'
+        super().__init__(prefix + ':P', **kwargs)
+
+    def move(self, position, **kwargs):
+        """
+        Move the gantry to a specific position
+
+        Calls OMMotor.move but first checks that the gantry is coupled before
+        proceeding. This is not a safety measure, but instead just here largely
+        for bookkeeping and to give the operator further feedback on why the
+        requested move is not completed.
+
+        Parameters
+        ----------
+        position: float
+            Requested position
+
+        kwargs:
+            Passed to OMMotor.move
+        """
+        # Check that the gantry is not decoupled
+        if self.decoupled.get():
+            raise PermissionError("The gantry is not currently coupled")
+        return super().move(position, **kwargs)
+
+
+class OffsetMirror(Device):
+    """
+    X-Ray offset mirror class for each individual mirror system used in the FEE
+    and XRT. Controls for the pitch, and primary gantry x and y motors are
+    included.
+
+    When controlling the pitch motor, if the piezo is set to 'PID' mode, then
+    the pitch mechanism is setup to first move the stepper as close to the
+    desired position, then the piezo will kick in to constantly try and correct
+    any positional changes.
+
+    Parameters
+    ----------
+    prefix : str
+        The EPICS base PV of the pitch motor
+
+    prefix_xy : str
+        The EPICS base PV of the gantry x and y gantry motors
+
+    xgantry_prefix : str
+        The name of the horizontal gantry if not identical to the prefix
+
+    read_attrs : sequence of attribute names, optional
+        The signals to be read during data acquisition (i.e., in read() and
+        describe() calls)
+
+    configuration_attrs : sequence of attribute names, optional
+        The signals to be returned when asked for the motor configuration (i.e.
+        in read_configuration(), and describe_configuration() calls)
+
+    name : str, optional
+        The name of the offset mirror
+
+    parent : instance or None, optional
+        The instance of the parent device, if applicable
+    """
+    # Pitch Motor
+    pitch = FC(Pitch, "MIRR:{self.prefix}")
+    # Gantry motors
+    xgantry = FC(Gantry, "{self._prefix_xy}:X",
+                 gantry_prefix="{self._xgantry}",
+                 add_prefix=['suffix', 'gantry_prefix'])
+    ygantry = FC(Gantry, "{self._prefix_xy}:Y")
+    # Transmission for Lightpath Interface
+    transmission = 1.0
+    SUB_STATE = 'sub_state_changed'
+
+    _default_read_attrs = ['pitch', 'xgantry.readback',
+                           'xgantry.gantry_difference']
+
+    _default_configuration_attrs = ['ygantry.setpoint']
+
+    def __init__(self, prefix, *, prefix_xy=None,
+                 xgantry_prefix=None, **kwargs):
```

Comment:
Design question: should we enforce `name` as a required `kwarg` here and at similar `__init__` statements, or ignore it and let the base `Device` class enfore the required `kwarg`?

Reply:
Leaving as a kwarg. Lets have Device handle this.